pkg/provider/aws/resources/eks.go

```diff
@@ -0,0 +1,212 @@
+// The method will create all of the corresponding IAM Roles necessary and attach all the execution units references to the following objects
+func CreateEksCluster(appName string, clusterName string, subnets []*Subnet, securityGroups []*any, units []*core.ExecutionUnit, dag *core.ResourceGraph) {
+	references := []core.AnnotationKey{}
+	for _, u := range units {
+		references = append(references, u.Provenance())
+	}
+
+	clusterRole := createClusterAdminRole(appName, clusterName+"-k8sAdmin", references)
+	dag.AddResource(clusterRole)
+
+	cluster := NewEksCluster(appName, clusterName, subnets, securityGroups, clusterRole)
+	cluster.ConstructsRef = references
+	dag.AddResource(cluster)
+	dag.AddDependency2(cluster, clusterRole)
+
+	fargateRole := createPodExecutionRole(appName, clusterName+"-FargateExecutionRole", references)
+	dag.AddResource(fargateRole)
+
+	profile := NewEksFargateProfile(cluster, subnets, fargateRole, references)
+	profile.Selectors = append(profile.Selectors, &FargateProfileSelector{Namespace: "default", Labels: map[string]string{"klotho-fargate-enabled": "true"}})
+
+	dag.AddResource(profile)
+	dag.AddDependency2(profile, fargateRole)
+	dag.AddDependency2(profile, cluster)
+
+	nodeRole := createNodeRole(appName, clusterName+"-NodeGroupRole", references)
+	dag.AddResource(nodeRole)
+
+	nodeGroup := NewEksNodeGroup(cluster, subnets, nodeRole, references)
+	dag.AddResource(nodeGroup)
+	dag.AddDependency2(nodeGroup, nodeRole)
+	dag.AddDependency2(nodeGroup, cluster)
+
+	for _, s := range subnets {
+		dag.AddDependency2(cluster, s)
+		dag.AddDependency2(nodeGroup, s)
+		dag.AddDependency2(profile, s)
+	}
+}
+
+func createClusterAdminRole(appName string, roleName string, refs []core.AnnotationKey) *IamRole {
+	clusterRole := NewIamRole(appName, roleName, refs, EKS_ASSUME_ROLE_POLICY)
+	clusterRole.AddAwsManagedPolicies([]string{"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"})
+	return clusterRole
+}
+
+func createPodExecutionRole(appName string, roleName string, refs []core.AnnotationKey) *IamRole {
+	fargateRole := NewIamRole(appName, roleName, refs, EKS_FARGATE_ASSUME_ROLE_POLICY)
+	fargateRole.InlinePolicy = &PolicyDocument{Version: VERSION, Statement: []StatementEntry{
+		{
+			Effect: "Allow",
+			Action: []string{
+				"logs:CreateLogStream",
+				"logs:CreateLogGroup",
+				"logs:DescribeLogStreams",
+				"logs:PutLogEvents",
+			},
+			Resource: []core.IaCValue{{Property: "*"}},
+		},
+	}}
+	return fargateRole
+}
+
+func createNodeRole(appName string, roleName string, refs []core.AnnotationKey) *IamRole {
+	nodeRole := NewIamRole(appName, roleName, refs, EC2_ASSUMER_ROLE_POLICY)
+	nodeRole.AddAwsManagedPolicies([]string{"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+		"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy", "arn:aws:iam::aws:policy/AWSCloudMapFullAccess",
+		"arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"})
```

Comment:
minor: I would add a newline after the opening `[]string{`, and then again after the last element (`CloudWatchAgentServerPolicy`, and also adding a comma after it as required by Go style). That way, if we add or remove items in the future, the diff will be cleaner.